pkg/cli/subcommands/check/command.go

```diff
@@ -141,9 +143,23 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 					LogParams:            logimpl.ForOneShot(globalParams.LoggerName, "off", true)}),
 				core.Bundle(),
 
+				// workloadmeta setup
+				collectors.GetCatalog(),
+				fx.Provide(func(config config.Component) workloadmeta.Params {
```

Comment:
I feel like we have this function logic now in different place, can/should we move it inside the workloadmeta package?

Reply:
Indeed we do but I think that the current pattern is to define parameters before providing a module and there might be slight differences in the different parameters depending on the command. 

[Here](https://github.com/search?q=repo%3ADataDog%2Fdatadog-agent+workloadmeta.Params&type=code) are the different places were we define workloadmeta params.

Here is were we have exactly the same logic https://github.com/DataDog/datadog-agent/blob/f425dfa882dd9ca8533172c246ea047be1a40799/cmd/agent/subcommands/run/command.go#L308-L320 so we wouldn't benefit a lot from a factorization.
- It looks like the pattern is to define the parameters before the module

I chose to follow the current pattern but I am happy to receive any other proposal.

Reply:
I think there is a solution that could work by defining default params and setting the params as optional. Will take a deeper look

Reply:
Only two flavors are supported here: clusteragent and agent. Does cluster check use this function? If not, we can either move this piece of code to workloadmeta or simplify the condition check by removing cluster flavor.

Reply:
CLC runners have the flavor `NodeAgent` as well. I think we should indeed move it somewhere.